src/main/java/com/avispl/symphony/dal/infrastructure/management/qsc/qsysreflect/utils/SystemResponseDeserializer.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * Copyright (c) 2022 AVI-SPL, Inc. All Rights Reserved.
+ */
+package com.avispl.symphony.dal.infrastructure.management.qsc.qsysreflect.utils;
+
+import java.io.IOException;
+
+import com.fasterxml.jackson.core.JsonParser;
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.fasterxml.jackson.databind.DeserializationContext;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.deser.std.StdDeserializer;
+
+import com.avispl.symphony.dal.infrastructure.management.qsc.qsysreflect.dto.SystemResponse;
+
+/**
+ * Custom Deserializer class for SystemResponse
+ *
+ * @author Duy Nguyen
+ * @version 1.0.0
+ * @since 1.0.0
+ */
+public class SystemResponseDeserializer extends StdDeserializer<SystemResponse> {
+
+	/**
+	 * SystemResponseDeserializer no arg constructor
+	 */
+	public SystemResponseDeserializer() {
+		this(null);
+	}
+
+	/**
+	 * SystemResponseDeserializer with arg constructor
+	 * @param vc Class name
+	 */
+	protected SystemResponseDeserializer(Class<?> vc) {
+		super(vc);
+	}
+
+	@Override
+	public SystemResponse deserialize(JsonParser jsonParser, DeserializationContext deserializationContext) throws IOException, JsonProcessingException {
+		JsonNode jsonNode = jsonParser.getCodec().readTree(jsonParser);
+		SystemResponse systemResponse = new SystemResponse();
+		systemResponse.setId(jsonNode.get("id").asInt());
+		systemResponse.setCode(jsonNode.get("code").asText());
+		systemResponse.setName(jsonNode.get("name").asText());
+		systemResponse.setStatusString(jsonNode.get("status").get("message").asText());
+		JsonNode detailsNode = jsonNode.get("status").get("details");
+		JsonNode itemsNode = detailsNode.get("items");
+		if (itemsNode != null) {
+			systemResponse.setNormalAlert(itemsNode.get("normal").asInt());
+			systemResponse.setWarningAlert(itemsNode.get("warning").asInt());
+			systemResponse.setFaultAlert(itemsNode.get("fault").asInt());
+			systemResponse.setUnknownAlert(itemsNode.get("unknown").asInt());
+		}
+		JsonNode designNode = jsonNode.get("design");
+		systemResponse.setDesignName(designNode.get("name").asText());
+		systemResponse.setDesignPlatform(designNode.get("platform").asText());
+		systemResponse.setUptime(designNode.get("uptime").asLong());
+		systemResponse.setCoreName(jsonNode.get("core").get("name").asText());
```

Comment:
can we guarantee that each of this nodes will be present at all times and NPE won't be triggered? I mean, are there any null-checks needed here?